### PR TITLE
Added support for launch angles of RF signal from source.

### PR DESCRIPTION
**Pull request by jflaherty13**
Added calculation and output of the launch angles of the RF signal from a source vertex.  The Ray Tracer natively calculated the receiving and launch angles with respect to the position vector of the receiver antenna (referred to as **_target_** in the code).  A diagram of the vector geometry can be seen in the image below:
image
where **_target_** is the position vector of the receiver antenna relative to the center of the Earth, **_source_** is the position vector of the source vertex relative to the center of the Earth.  As implied by the names, _**launchVector**_ and _**receiveVector**_ are the trajectories of the RF signal at the source and receiver, respectively.  Within the Ray Tracer, there are values for _**launchAngle**_ and **_receiptAngle_**, which are measured with respect the the _**target**_ vector as shown in the figure.

The ray tracing table script uses the _**receiveVector**_ to calculate _**arrivalTheta**_ and **_arrivalPhi_** of the RF trajectory with the z-axis along the Earth's radius and stores those values in the ray tracing tables.  This push uses that same logic to calculate the **_launchTheta_** and **_launchPhi_** of the launch vector and store them into the ray tracing tables.

### CODE REVIEW

**Comment by clark2668** [comment]
This repeats a question I asked on the AraSim PR, but is launch really calculated wrt to target, and not emitter? The emitter seems so much more obvious.

**Comment by jflaherty13** [comment]
> This repeats a question I asked on the AraSim PR, but is launch really calculated wrt to target, and not emitter? The emitter seems so much more obvious.

Yes this threw me for a loop.  The ray tracer calculates both the launch and arrival angles with respect to the target.  I didn't want to change that convention as that would risk affecting existing code that uses the launch angle.

**Comment by clark2668** [comment]
How odd. But okay then, looks alright then.

File: AraCorrelator/RayTraceCorrelator.h [hunk 1]
```diff
@@ -117,6 +119,22 @@ class RayTraceCorrelator : public TObject
             double &arrivalTheta, double &arrivalPhi
         );
         
+        //! function to get lookup the launch angle information
+        /*!
+            \param ant antenna index
+            \param solNum which solution number (0 = direct, 1 = reflected/refracted)
+            \param thetaBin the theta bin desired (bin space, not angle space!!)
+            \param phiBin the phi bin desired (bin space, not angle space!!)
+            \param launchTheta passed by reference, content is replaced with the launch theta angle (in a coordinate where z-axis is along the Earth's radius)
```
Comment (clark2668) [inline AraCorrelator/RayTraceCorrelator.h:128]: Advise adding clarifying doc that launch angle is wrt to target, not emitter. Otherwise, looks good to me.

**Review by clark2668 — CHANGES_REQUESTED** [review]

File: AraCorrelator/RayTraceCorrelator.h (hunk 1 above)
Comment (jflaherty13) [reply AraCorrelator/RayTraceCorrelator.h:128]: The angles in this function are in the station-centric coordinates, not with respect to the target vector.  The launch angle is initially calculated w.r.t. the target position vector in the [RaySolver class in AraSim](https://github.com/ara-software/AraSim/blob/2d7de9db0fa2231a23e487754d8e9b72f9b2f606/RaySolver.cc#L870).  This function uses the same logic as Report::GetAngleAnt, which gets the arrival angles in station-centric coordinates, to get the launch angles in station-centric coordinates.

Comment (clark2668) [reply AraCorrelator/RayTraceCorrelator.h:128]: Okay, I see. Maybe make a comment to that effect in the code? But otherwise looks ok.

**Review by clark2668 — APPROVED** [review]

**Pull request merged**